File: src/main/kotlin/ch/bildspur/seqosc/net/OSCClient.kt
package ch.bildspur.seqosc.net

import java.net.DatagramPacket
import java.net.DatagramSocket
import java.net.InetAddress

class OSCClient {

    val socket = DatagramSocket()

    init {
        socket.broadcast = true
    }

    fun send(address : InetAddress, port : Int, packet : OSCPacket) {
        val datagramPacket = DatagramPacket(packet.data, 0, packet.data.size, address, port)
        socket.send(datagramPacket)
    }
}

File: dotnet/net/OSCClient.cs
using System.Net;
using System.Net.Sockets;

namespace SeqOSC.Net
{
    public class OSCClient
    {
        public UdpClient Client { get; private set; } = new UdpClient();

        public void Send(IPEndPoint endPoint, OSCPacket packet)
        {
            Client.Send(packet.Data, packet.Data.Length, endPoint);
        }
    }
}